### PR TITLE
Hapi17

### LINKED ISSUE #36: hapi v17 support

### PR DESCRIPTION
**Pull request by danielb2**
* Fixes #36 
* Updates dependencies
* version bump

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,5 +2,4 @@ sudo: false
 language: node_js
 
 node_js:
-  - "4"
-  - "6"
+  - "8"
```
Comment (AdriVanHoudt) [inline .travis.yml]: I would add "node" and/or "9" to be future proof

Comment (Marsup) [reply .travis.yml]: Having both is the unofficial guidelines all hapi projects should follow.

File: lib/index.js [hunk 2]
```diff
@@ -11,10 +11,9 @@ require('useragent/features');                      // Enhances Useragent
 const internals = {};
 
 
-exports.register = (server, options, next) => {
+exports.register = async function (server, options) {
 
-    server.ext('onRequest', internals.onRequest);
-    return next();
+    await server.ext('onRequest', internals.onRequest);
```
Comment (Marsup) [inline lib/index.js]: `server.ext` is not async, so you don't need to await, and you don't need register to be async.

File: lib/index.js [hunk 3]
```diff
@@ -23,8 +22,11 @@ exports.register.attributes = {
 };
 
 
-internals.onRequest = (request, reply) => {
+exports.name = 'scooter';
```
Comment (Marsup) [inline lib/index.js]: Shouldn't you just export pkg like most other plugins do ?

**Comment by danielb2** [comment]
@Marsup fixed

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,1809 @@
+{
```
Comment (Marsup) [inline package-lock.json]: No package-log, add it to gitignore.

**Comment by danielb2** [comment]
@marsup  can check again ?

**Comment by Marsup** [comment]
Ready to go I think.

**Comment by Marsup** [comment]
Just noticed the readme is not really up to date with hapi 17.

**Comment by traviswimer** [comment]
When I `npm install` scooter 4.0.0 it doesn't seem to have any of the updates for hapi 17. I might just have something weird going on locally, but I think maybe the wrong code was published to npm. Anyone know?

**Pull request merged**